src/TextHandler.mss

```diff
@@ -111,27 +112,39 @@ function FiguredBassTextHandler (this, textObject) {
 }  //$end
 
 
-function CreateAnchoredText (this, textObj) {
+function CreateAnchoredText (this, textObject) {
     anchoredText = libmei.AnchoredText();
-    AddFormattedText(anchoredText, textObj);
+    AddFormattedText(anchoredText, textObject);
     return anchoredText;
 }  //$end
 
+function CreateDirective (this, textObject) {
+    directive = GenerateControlEvent(textObject, 'Dir');
+    AddFormattedText(directive, textObject);
+    styleIdPrefix = 'text.staff.';
+    text_style = Substring(textObject.StyleId, Length(styleIdPrefix));
```

Comment:
This should probably be `textStyle` since camelCase is generally the style here.